DOCSP-34971: Update minimum MongoDB version to v5. (#722)
mongodb/docs-app-services: source/includes/mongodb-5.0-required-for-sync-admonition.rst

.. note:: MongoDB Version 5.0 or Higher Required
   

>    In order to use Atlas Device Sync, your Atlas cluster must use MongoDB version 5.0 or higher.
>    When setting up your cluster, select :guilabel:`MongoDB 5.0` (or a higher version)

Reviewer: Some minor edits for the pre-existing content. Feel free to ignore if you want to just merge this.

```suggestion
   Atlas Device Sync requires an Atlas cluster running MongoDB version 5.0 or later.
   When setting up your cluster, select :guilabel:`MongoDB 5.0` (or a later version)
```
